py/DREAM/Output/SPIShardPositions.py

```diff
@@ -59,6 +125,105 @@ def calcRadialCoordinate(self, shards=None, t=None):
         thetap = np.arctan2(yp[t,shards],xp[t,shards])
 
         return rhop, thetap
+
+
+    def plotAtTime(self, t=-1, shards=None, ax=None, show=None):
+        """
+        Plot the pellet shards over the poloidal cross-section at a given time.
+        """
+        black = (87/255, 117/255, 144/255)
+        red = (249/255, 65/255, 68/255)
+
+        genax = ax is None
+
+        if genax:
+            ax = plt.axes()
+
+            if show is None:
+                show = True
+
+        if shards is None:
+            shards = slice(None)
+
+        if 'eq' not in self.grid:
+            raise OutputException("Cannot plot poloidal trajectory when equilibrium data is not stored in output.")
+
+        eq = self.grid.eq
+
+        xp = self.data[:,0::3,0]
+        yp = self.data[:,1::3,0]
+
+        eq.visualize(ax=ax, shifted=True, maxis=False)
+
+        ax.plot(xp[0,0], yp[0,0], 'o', color=red)
```

Comment:
Once again, do the shards always start in the same position initially?

Reply:
In the first simulation, yes. In contrast to your previous comment though, here I don't see any easy way of calculating the actual origin (because the pellet shatter-point is (sadly) not stored in the settings file).

Reply:
Could we maybe have an if-statement, such that if the shards don't start in the same location, we can just print a warning about there being no information about the origin and skip the plotting of the origin, instead of assuming it starts in the first shards initial position?